test/LEGO.AsyncAPI.Tests/Bindings/Sns/SnsBindings_Should.cs

```diff
@@ -0,0 +1,57 @@
+using System;
+
+namespace LEGO.AsyncAPI.Tests.Bindings.Sns
+{
+    using NUnit.Framework;
+    using System.Collections.Generic;
+    using FluentAssertions;
+    using LEGO.AsyncAPI.Models;
+    using LEGO.AsyncAPI.Models.Bindings.Sns;
+    using LEGO.AsyncAPI.Readers;
+
+    internal class SnsBindings_Should
+    {
+        [Test]
+        public void SnsChannelBinding_WithFilledObject_SerializesAndDeserializes()
+        {
+            // Arrange
+            var expected = 
+                @"bindings:
+  sns:
+    name: myTopic
+    policy:
+      statements:
+        - principal: hello";
+
+            var channel = new AsyncApiChannel();
+            channel.Bindings.Add(new SnsChannelBinding()
+            {
+                Name = "myTopic",
+                Policy = new Policy()
+                {
+                    Statements = new List<Statement>()
+                    {
+                        new Statement()
+                        {
+                            Principal = "hello",
+                        },
+                    },
+                },
+            });
+            
+            // Act
+            var actual = channel.SerializeAsYaml(AsyncApiVersion.AsyncApi2_0);
+            
+            // Assert
+            actual = actual.MakeLineBreaksEnvironmentNeutral();
+            expected = expected.MakeLineBreaksEnvironmentNeutral();
+            
+            var binding = new AsyncApiStringReader().ReadFragment<AsyncApiChannel>(actual, AsyncApiVersion.AsyncApi2_0, out _);
+            
+            // Assert
+            Assert.AreEqual(actual, expected);
+            binding.Should().BeEquivalentTo(channel);
```

Comment:
@VisualBean Question: The test is serializing the binding correctly here but failing to deserialize it, the binding dictionary is coming back empty, so somewhere the reader isn't picking up the `sns` binding, but I'm not sure what I've done wrong. Do you have any intuition on this? Is there another place I need to register to get the reader to recognise `sns`?

Reply:
You will need to add the SNS type to the AsyncApiChannelBindingDeserializer
Currently it relies on the BindingType enum to figure out which deserialiser to use 
```csharp
case BindingType.Kafka:
                    return LoadBinding("ChannelBinding", property.Value, kafkaChannelBindingFixedFields);
```

Reply:
This is what im hoping to push together in the bindings project. having bindings as a singular source of read and write..

Good work so far!

Reply:
I am about ready to release the bindings project. Just need to finalize the last bit of bindings and how to resolve schemas, but otherwise we should be good to go.

Reply:
Thanks for the guidance and updates. I’ll rebase from there once it’s merged. Looks like it will be a big improvement and work saver for getting everything set up. I’m back tomorrow so will have some time to sync up with it all and make more progress.

Reply:
The change is in main now